mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -102,33 +141,87 @@ class Completions(
   inline private def undoBacktick(label: String): String =
     label.stripPrefix("`").stripSuffix("`")
 
-  private def findSuffix(methodSymbol: Symbol): Option[String] =
-    if shouldAddSnippet && methodSymbol.is(Flags.Method) then
-      lazy val extensionParam = methodSymbol.extensionParam
-
-      val paramss =
-        if methodSymbol.is(Flags.Extension) then
-          methodSymbol.paramSymss.filter(
-            !_.contains(extensionParam)
-          )
-        else methodSymbol.paramSymss
-
-      paramss match
-        case Nil => None
-        case List(Nil) => Some("()")
-        case _ if config.isCompletionSnippetsEnabled =>
-          val onlyParameterless = paramss.forall(_.isEmpty)
-          lazy val onlyImplicitOrTypeParams = paramss.forall(
-            _.exists { sym =>
-              sym.isType || sym.is(Implicit) || sym.is(Given)
-            }
-          )
-          if onlyParameterless then Some("()" * paramss.length)
-          else if onlyImplicitOrTypeParams then None
-          else Some("($0)")
-        case _ => None
-      end match
-    else None
+  private def getParams(symbol: Symbol) =
+    lazy val extensionParam = symbol.extensionParam
+    if symbol.is(Flags.Extension) then
+      symbol.paramSymss.filter(
+        !_.contains(extensionParam)
+      )
+    else symbol.paramSymss
+
+  private def findSuffix(symbol: Symbol): Option[String] =
+    val bracesSuffix =
+      if shouldAddSnippet && symbol.is(
+          Flags.Method
+        ) && !isTypePosition
+      then
+        val paramss = getParams(symbol)
+        paramss match
+          case Nil => ""
+          case List(Nil) => "()"
+          case _ if config.isCompletionSnippetsEnabled =>
+            val onlyParameterless = paramss.forall(_.isEmpty)
+            lazy val onlyImplicitOrTypeParams = paramss.forall(
+              _.exists { sym =>
+                sym.isType || sym.is(Implicit) || sym.is(Given)
+              }
+            )
+            if onlyParameterless then "()" * paramss.length
+            else if onlyImplicitOrTypeParams then ""
+            else "($0)"
+          case _ => ""
+        end match
+      else ""
+
+    // symbol.info.typeParams.foreach(param =>
+    //   pprint.log(param.paramInfo)
+    // )
+
+    val bracketSuffix =
+      if shouldAddSnippet
+        && (isTypePosition || isNewPosition || isInstantiationOrMethodCallPos)
```

Comment:
```suggestion
     
    val shouldAddSquareBracket = path match {
      case (_ : Ident) :: (_ : AppliedTypeTree ) :: _ => false
      case (i : Ident):: _ =>
        val span = i.srcPos.span
        pprint.log(text(span.end))
        if (span.exists){ // TODO check spaces/tabs
          text(span.end) != '['
        } else false
      case _ => false
    }

    val bracketSuffix =
      if shouldAddSnippet && shouldAddSquareBracket
        && (isTypePosition || isNewPosition || isInstantiationOrMethodCallPos)
```

seems the ArrayBuffer[Int] is not seens as TypeApply, which is why I think we need to do a workaround. Something like the above